nifi-extension-bundles/nifi-standard-bundle/nifi-standard-rules/src/main/java/org/apache/nifi/flowanalysis/rules/RestrictBackpressureSettings.java

```diff
@@ -0,0 +1,242 @@
+                .collect(Collectors.toMap(c -> c.getIdentifier(), Function.identity()));
+
+        pg.getConnections().stream().forEach(
+                connection -> {
+                    if (connection.getBackPressureObjectThreshold() < minCount) {
+                        results.add(buildViolation(connection,
+                                idComponent.get(connection.getSource().getId()),
+                                idComponent.get(connection.getDestination().getId()),
+                                BackpressureViolationType.BP_COUNT_THRESHOLD_BELOW_LIMIT,
+                                getViolationMessage(BackpressureViolationType.BP_COUNT_THRESHOLD_BELOW_LIMIT, connection.getBackPressureObjectThreshold().toString(), Long.toString(minCount))));
+                    }
+                    if (connection.getBackPressureObjectThreshold() > maxCount) {
+                        results.add(buildViolation(connection,
+                                idComponent.get(connection.getSource().getId()),
+                                idComponent.get(connection.getDestination().getId()),
+                                BackpressureViolationType.BP_COUNT_THRESHOLD_ABOVE_LIMIT,
+                                getViolationMessage(BackpressureViolationType.BP_COUNT_THRESHOLD_ABOVE_LIMIT, connection.getBackPressureObjectThreshold().toString(), Long.toString(maxCount))));
+                    }
+                    final double sizeThreshold = DataUnit.parseDataSize(connection.getBackPressureDataSizeThreshold(), DataUnit.B);
+                    if (Double.compare(sizeThreshold, minSize) < 0) {
+                        results.add(buildViolation(connection,
+                                idComponent.get(connection.getSource().getId()),
+                                idComponent.get(connection.getDestination().getId()),
+                                BackpressureViolationType.BP_SIZE_THRESHOLD_BELOW_LIMIT,
+                                getViolationMessage(BackpressureViolationType.BP_SIZE_THRESHOLD_BELOW_LIMIT, connection.getBackPressureDataSizeThreshold(), context.getProperty(SIZE_MIN).getValue())));
+                    }
+                    if (Double.compare(sizeThreshold, maxSize) > 0) {
+                        results.add(buildViolation(connection,
+                                idComponent.get(connection.getSource().getId()),
+                                idComponent.get(connection.getDestination().getId()),
+                                BackpressureViolationType.BP_SIZE_THRESHOLD_ABOVE_LIMIT,
+                                getViolationMessage(BackpressureViolationType.BP_SIZE_THRESHOLD_ABOVE_LIMIT, connection.getBackPressureDataSizeThreshold(), context.getProperty(SIZE_MAX).getValue())));
+                    }
+                }
+            );
+
+        return results;
+    }
+
+    private GroupAnalysisResult buildViolation(final VersionedConnection connection, final VersionedComponent source,
```

Comment:
This method chooses whether to assign the violation to the connection or nearest real Processor.  Ideally, backpressure violations would always be assigned to the connection.  Was this done as a workaround for NIFI-13953 because the UI does not showing a "Go To" link for connection violations?

I'm not sure we should keep a workaround like this.  If you feel that we need the workaround, then a comment in the code to that effect should be here, stating why it works like this and when it should be replaced.

Reply:
Correct - I chose this approach because of [NIFI-13953](https://issues.apache.org/jira/browse/NIFI-13953) as the "Go To" is a very important aspect when dealing with very large flows. I'll add a TODO and explanation in the code to revisit this.

Reply:
My bad, it's been a bit too long since I did this. The reason why we want the violation to be on the processor when we have one (either as source or destination) is because in case the rule is "enforced" we want the corresponding component to be invalid.

Reply:
Understood, thanks!